### ISSUE TITLE
Add social logins

### ISSUE THREAD

**Issue by techno-disaster** [bartekpacia/mentorship-client#137]
## Description
As a user I would like to be able to login/sign-up using a social account, so I don't need to create a profile from scratch

### Update [Required]
- [ ] add google and apple sign in

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

**Comment by techno-disaster** [comment]
Status on hold untill merged with backend.

**Comment by shivanigupta19** [comment]
Please assign me this issue

**Comment by techno-disaster** [comment]
@shivanigupta19  sure, if you want to start working on this please merge Yugantars social logins pr found [here](https://github.com/anitab-org/mentorship-backend/pull/719) . You can ask doubts here, incase you are stuck.

**Comment by shivanigupta19** [comment]
> @shivanigupta19 sure, if you want to start working on this please merge Yugantars social logins pr found [here](https://github.com/anitab-org/mentorship-backend/pull/719) . You can ask doubts here, incase you are stuck.

Sorry by mistake i unassigned , please assign me again

**Comment by shivanigupta19** [comment]
> @shivanigupta19 sure, if you want to start working on this please merge Yugantars social logins pr found [here](https://github.com/anitab-org/mentorship-backend/pull/719) . You can ask doubts here, incase you are stuck.

Sir , i have windows so , i cannot implement apple sign in the project.
you are not using firebase in this project

**Comment by techno-disaster** [comment]
> > @shivanigupta19 sure, if you want to start working on this please merge Yugantars social logins pr found [here](https://github.com/anitab-org/mentorship-backend/pull/719) . You can ask doubts here, incase you are stuck.
> 
> Sir , i have windows so , i cannot implement apple sign in the project.
> you are not using firebase in this project

I personally have only used google sign in up untill today, @yugantarjain  can you confirm if implementing apple sign in requires a mac? IMO it shoudn't
@shivanigupta19 can you implement google sign in till then?

**Comment by shivanigupta19** [comment]
> > > @shivanigupta19 sure, if you want to start working on this please merge Yugantars social logins pr found [here](https://github.com/anitab-org/mentorship-backend/pull/719) . You can ask doubts here, incase you are stuck.
> > 
> > 
> > Sir , i have windows so , i cannot implement apple sign in the project.
> > you are not using firebase in this project
> 
> I personally have only used google sign in up untill today, @yugantarjain can you confirm if implementing apple sign in requires a mac? IMO it shoudn't
> @shivanigupta19 can you implement google sign in till then?

Sure but in back-end what are you using ?

**Comment by techno-disaster** [comment]
@yashasvimisra2798 We have a flask backend setup. Social logins are not merged yet so you will have to use [this](https://github.com/yugantarjain/mentorship-backend/tree/social-sign-in) branch.

**Comment by shivanigupta19** [comment]
@Techno-Disaster should i write your email in Support email Shown on the consent screen for user support.
if Yes , please provide me your email- id.

**Comment by shivanigupta19** [comment]
@Techno-Disaster  please provide me google account with password to proceed this task.

**Comment by techno-disaster** [comment]
> @Techno-Disaster please provide me google account with password to proceed this task.

any google account should work ig?

**Comment by shivanigupta19** [comment]
> > @Techno-Disaster please provide me google account with password to proceed this task.
> 
> any google account should work ig?

@Techno-Disaster it will better if i use project's google account so that if there is any email regarding project or any changes , ll be easy to communicate but if i use my one , so it will create issues in future so , i cannot use my personal email. so , please make account if there is not.

**Comment by techno-disaster** [comment]
> > > @Techno-Disaster please provide me google account with password to proceed this task.
> > 
> > 
> > any google account should work ig?
> 
> @Techno-Disaster it will better if i use project's google account so that if there is any email regarding project or any changes , ll be easy to communicate but if i use my one , so it will create issues in future so , i cannot use my personal email. so , please make account if there is not.

There's no such project mail id as of now, please use any google account or create one. We will be communicating updates only here or on zulip.

**Comment by shivanigupta19** [comment]
> > > > @Techno-Disaster please provide me google account with password to proceed this task.
> > > 
> > > 
> > > any google account should work ig?
> > 
> > 
> > @Techno-Disaster it will better if i use project's google account so that if there is any email regarding project or any changes , ll be easy to communicate but if i use my one , so it will create issues in future so , i cannot use my personal email. so , please make account if there is not.
> 
> There's no such project mail id as of now, please use any google account or create one. We will be communicating updates only here or on zulip.

sure , i ll make a new one.

**Comment by shivanigupta19** [comment]
@Techno-Disaster please provide me Application Homepage link , Application Privacy Policy link , Authorized domains

**Comment by techno-disaster** [comment]
> @Techno-Disaster please provide me Application Homepage link , Application Privacy Policy link , Authorized domains

Not sure why you might need these or what these are, but i guess you mean these?
https://mentorship-backend-temp.herokuapp.com/
We don't have a privacy policy link or auth domains afaik. 
Could you tell me why / where this is needed?

**Comment by shivanigupta19** [comment]
@Techno-Disaster need to fill this.
![WhatsApp Image 2020-09-02 at 3 35 54 PM](https://user-images.githubusercontent.com/53371471/91968493-58ab7200-ed32-11ea-8713-334f8285adef.jpeg)

**Comment by techno-disaster** [comment]
> @Techno-Disaster need to fill this.
> ![WhatsApp Image 2020-09-02 at 3 35 54 PM](https://user-images.githubusercontent.com/53371471/91968493-58ab7200-ed32-11ea-8713-334f8285adef.jpeg)

Never seen this before, could you tell me what this is and share the link?. But if the fields are not mandatory ignore them for now

**Comment by techno-disaster** [comment]
Hey, @shivanigupta19 any updates on this issue?

**Comment by techno-disaster** [comment]
Unassigning @shivanigupta19  because of no updates. Feel free to ping me i case you want to be assigned again

**Comment by techno-disaster** [comment]
@anitab-org/coding-team Anyone want to pick this up?

**Comment by NarimaneHennouni** [comment]
Hi, can I be assigned to this?

**Comment by techno-disaster** [comment]
@NarimaneHennouni sure you can start working on it right away

**Comment by NarimaneHennouni** [comment]
@Techno-Disaster okay thank you

**Comment by NarimaneHennouni** [comment]
@Techno-Disaster Hi, I was wondering if I am allowed to use Firebase for Google and Apple sign in?

**Comment by techno-disaster** [comment]
> @Techno-Disaster Hi, I was wondering if I am allowed to use Firebase for Google and Apple sign in?

No, we already have a backend PR up. If you merge that and host the baco locally you should see the social media sign in endpoints

**Comment by NarimaneHennouni** [comment]
> > @Techno-Disaster Hi, I was wondering if I am allowed to use Firebase for Google and Apple sign in?
> 
> No, we already have a backend PR up. If you merge that and host the baco locally you should see the social media sign in endpoints

So I only need to work on the UI?

**Comment by techno-disaster** [comment]
@naman1901 No, host the mentorship backend locally, merge the social logins branch linked above and  integrate the google and apple login in the flutter app

**Comment by techno-disaster** [comment]
@NarimaneHennouni any updates on this issue?

**Comment by techno-disaster** [comment]
Unassigning due to no response. Please feel free to be assigned to this issue once again if you start working on it or are stuck somewhere.

**Comment by PaRaDoX50** [comment]
@Techno-Disaster Can you please assign this to me?

**Comment by techno-disaster** [comment]
> @Techno-Disaster Can you please assign this to me?

Sure.

**Comment by PaRaDoX50** [comment]
@Techno-Disaster Will I need to merge the Yugantar's social-sign-in branch with the current develop branch and use it as a backend locally or can I just directly use Yugantar's social-sign-in branch ? In the first case, there are a lot of conflicts which I will have to resolve.

**Comment by techno-disaster** [comment]
> @Techno-Disaster Will I need to merge the Yugantar's social-sign-in branch with the current develop branch and use it as a backend locally or can I just directly use Yugantar's social-sign-in branch ? In the first case, there are a lot of conflicts which I will have to resolve.

If you can, it would be better to merges Yugantar's branch locally with the current develop to get the backend up to date.
